modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/BookmarksFragment.kt

```diff
@@ -44,21 +81,198 @@ class BookmarksFragment : BaseFragment() {
 }
 
 @Composable
-private fun BookmarksPage(
+fun BookmarksPage(
     playerViewModel: PlayerViewModel,
+    bookmarksViewModel: BookmarksViewModel = hiltViewModel(),
 ) {
+    val state by bookmarksViewModel.uiState.collectAsStateWithLifecycle()
     val listData = playerViewModel.listDataLive.asFlow().collectAsState(initial = null)
     listData.value?.let {
-        Box(
-            contentAlignment = Alignment.Center,
+        val backgroundColor = Color(it.podcastHeader.backgroundColor)
+        when (state) {
+            is UiState.Empty,
+            is UiState.Loading,
+            -> Unit
+
+            is UiState.Loaded -> Content(
+                bookmarks = (state as UiState.Loaded).bookmarks,
+                backgroundColor = backgroundColor,
+            )
+        }
+        LaunchedEffect(Unit) {
+            bookmarksViewModel.loadBookmarks(
+                episodeUuid = it.podcastHeader.episodeUuid
+            )
+        }
+    }
+}
+
+@Composable
+private fun Content(
+    bookmarks: List<Bookmark>,
+    backgroundColor: Color,
+) {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = backgroundColor)
```

Comment:
Should we set this earlier, maybe as the background to the fragment as I'm seeing a white page flash while the content is loading.

<img width="580" alt="Screenshot 2023-07-05 at 10 41 19 am" src="https://github.com/Automattic/pocket-casts-android/assets/308331/9b7a0901-5ecb-4f88-aa3e-2f6e9b3b5e3d">

Reply:
It takes time for the view to load. I think it should be resolved when I add the layout for `Loading` and `Empty` states.